web/oauth.go

```diff
@@ -273,7 +273,11 @@ func completeOAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	state := r.URL.Query().Get("state")
 
-	uri := c.GetSiteURLHeader() + "/signup/" + service + "/complete"
+	siteURL := *c.App.Config().ServiceSettings.SiteURL
```

Comment:
Full disclosure: I'm not super familiar with this flow

Wondering if this single change is enough since there are quite a few other places in this file that seemingly use `c.GetSiteURLHeader`. Probably need to update all of them?